src/scribe_data/checkquery.py

```diff
@@ -0,0 +1,442 @@
+#!/usr/bin/env python3
+
+"""Command line tool for testing SPARQl queries against an endpoint."""
+
+import argparse
+import math
+import os
+import subprocess
+import sys
+import time
+import urllib.request
+from dataclasses import dataclass
+from http import HTTPStatus
+from pathlib import Path
+from typing import Optional
+from urllib.error import HTTPError
+
+import SPARQLWrapper as SPARQL
+from SPARQLWrapper import SPARQLExceptions
+from tqdm.auto import tqdm
+
+
+EXIT_SUCCESS = 0
+EXIT_FAILURE = 1
+EXIT_CLI_ERROR = 2
+
+PROJECT_ROOT = "Scribe-Data"
+
+
+@dataclass(repr=False, frozen=True)
+class QueryFile:
+    """Holds a reference to a file containing a SPARQL query."""
+
+    path: Path
+
+    def load(self, limit: int) -> str:
+        """Load the SPARQL query from 'path' into a string.
+
+        Args:
```

Comment:
Let's discuss how we want docstrings to work in terms of eventual documentation, m-charlton :) I added a point to the dev sync just now that we should look  into readthedocs or something for that. In that case we should standardize how we're doing these. I personally have no preferece. If memory serves me I went with the NumPy style, but then readability wise I like yours more! Just a question of whether they'd translate to a documentation web interface.